documentation/modules/snip-provider-names.adoc

```diff
@@ -0,0 +1,7 @@
+:_content-type: SNIPPET
+
+[NOTE]
+====
+Provider names cannot include the following character: +
+`.`
+====
```

Comment:
You missed starting with an alphabet char limitation.
Meaning that it should be something like:
"Provider names **must start with an alphabet char** and cannot include the following character: `.`"

(the provider names validation is a bit different than mapping and plans since it blocks starting with a alphanumeric char as well)